src/puppetlabs/puppetdb/scf/storage.clj

```diff
@@ -519,30 +520,45 @@
     (update-in resource [:tags] sutils/to-jdbc-varchar-array)
     resource))
 
+(s/defn id->certname :- String
+  [certname-id :- Long]
+  (->> (query-to-vec "select certname from certnames where id=?" certname-id)
+       first
+       :certname))
+
 (s/defn insert-catalog-resources!
   "Returns a function that accepts a seq of ref keys to insert"
   [certname-id :- Long
+   certname :- String
    refs-to-hashes :- {resource-ref-schema String}
    refs-to-resources :- resource-ref->resource-schema]
   (fn [refs-to-insert]
-    {:pre [(every? resource-ref? refs-to-insert)]}
-
-    (update! (:catalog-volatility performance-metrics) (count refs-to-insert))
-
-    (insert-records*
-     :catalog_resources
-     (map (fn [resource-ref]
-            (let [{:keys [type title exported parameters tags file line] :as resource} (get refs-to-resources resource-ref)]
-              (convert-tags-array
-               {:certname_id certname-id
-                :resource (sutils/munge-hash-for-storage (get refs-to-hashes resource-ref))
-                :type type
-                :title title
-                :tags tags
-                :exported exported
-                :file file
-                :line line})))
-          refs-to-insert))))
+    (let [record (atom nil)]
+      {:pre [(every? resource-ref? refs-to-insert)]}
+      (update! (:catalog-volatility performance-metrics) (count refs-to-insert))
+      (try
+        (insert-records*
+          :catalog_resources
+          (map (fn [resource-ref]
+                 (let [{:keys [type title exported
+                               parameters tags file line] :as resource} (get refs-to-resources resource-ref)]
+                   (reset! record resource)
```

Comment:
Q: what do we think of this approach for ensuring we have the file/line of the failing resource?